pkgs/by-name/fi/filen-cli/package.nix

```
@@ -0,0 +1,85 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  file,
+}:
+
+stdenv.mkDerivation rec {
+  owner = "FilenCloudDienste";
+  pname = "filen-cli";
+  version = "0.0.19";
+
+  # Define platform-specific source URL
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${pname}-v${version}-linux-x64";
+        sha256 = "sha256-8y58HV82uDGhHDziCyhwcHaecactY331Ztbw+uq7oh4=";
+      }
+    else if stdenv.hostPlatform.system == "aarch64-linux" then
+      fetchurl {
+        url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${pname}-v${version}-linux-arm64";
+        sha256 = "sha256-Ee8MnwTKlHvZzK7PHoE9xft4WeeAw9qTnjk/1dcyhMY=";
+      }
+
+    else if stdenv.hostPlatform.system == "x86_64-darwin" then
+      fetchurl {
+        url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${pname}-v${version}-macos-x64";
+        sha256 = "sha256-J721M6VE74iZYavp9VpzpMQSMsBB4fAlu7djgceMWE4=";
+      }
+
+    else if stdenv.hostPlatform.system == "x86_64-windows" then
+      fetchurl {
+        url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${pname}-v${version}-win-x64";
+        sha256 = "sha256-ra2qbveLypV2ztDobhhOSneD3ORkuCqNy7i0k6K/7yg=";
+      }
+
+    else
+      throw "Unsupported platform: ${stdenv.hostPlatform.system}";
+
```

```suggestion
  nativeBuildInputs = [ autoPatchelfHook ];
```
Otherwise the binary will not run properly.